src/app/BUILD.gn

```diff
@@ -182,29 +176,55 @@ static_library("app") {
     "MessageDef/TimedRequestMessage.cpp",
     "MessageDef/WriteRequestMessage.cpp",
     "MessageDef/WriteResponseMessage.cpp",
-    "OTAUserConsentCommon.h",
+  ]
+
+  deps = [
+    ":app_config",
+    ":path-params",
+    ":revision_info",
+    "${chip_root}/src/lib/core",
+    "${chip_root}/src/lib/support",
+    "${chip_root}/src/protocols/interaction_model",
+    "${chip_root}/src/protocols/secure_channel",
+  ]
+}
+
+config("config-controller-dynamic-server") {
+  defines = [
```

Comment:
is this from existing logic or could we use a config_header? config_header is generally more friendly to people integrating into alternate build systems, however they will not propagate without explicit inclusion.

Reply:
I'd rather not include that in the refactor since it is a bit outside the scope of the refactor in this PR.
Using `config` makes the BUILD.gn cleaner while keeping the same structure of the intitial code.

Reply:
ok. These were existing defines.

We will have to eventually update them.